File: shoebill_plugins/LtrpApi/src/main/java/lt/ltrp/object/ContractJob.java
package lt.ltrp.object;

import java.util.Collection;
import java.util.List;

/**
 * @author Bebras
 *         2016.04.14.
 */
public interface ContractJob extends Job {

    void setRanks(List<ContractJobRank> ranks);
    int getMaxPaycheck();
    void setMaxPaycheck(int maxPaycheck);
    int getMinPaycheck();
    void setMinPaycheck(int minPaycheck);
    int getContractLength();
    void setContractLength(int contractLength);

    @Override
    Collection<ContractJobRank> getRanks();

    @Override
    ContractJobRank getRank(int id);

    Collection<JobVehicle> getVehicles(ContractJobRank rank);
}

File: shoebill_plugins/LtrpApi/src/main/java/lt/ltrp/job/object/ContractJob.kt
package lt.ltrp.job.`object`;

/**
 * @author Bebras
 *         2016.04.14.
 */
interface ContractJob: Job {

    override val ranks: MutableList<out ContractJobRank>
    var maxPaycheck: Int
    var contractLength: Int

    override fun getRankByNumber(number: Int): ContractJobRank?
    override fun getRankByUUID(uuid: Int): ContractJobRank?
    fun getVehicles(rank: ContractJobRank): Collection<JobVehicle>
}
